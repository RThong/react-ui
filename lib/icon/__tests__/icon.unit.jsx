import renderer from 'react-test-renderer'
import Icon from '../icon'
import React from 'react'
import { mount } from 'enzyme'
describe('icon', () => {
	it('渲染 传入props', () => {
		const json = renderer.create(<Icon type="alipay" />).toJSON()
		expect(json).toMatchSnapshot()
	})
	it('渲染 未传入props', () => {
		const json = renderer.create(<Icon />).toJSON()
		expect(json).toMatchSnapshot()
	})
	it('onClick', () => {
		const fn = jest.fn()
		const component = mount(<Icon type="alipay" onClick={fn} />)
		// 模拟click
		component.simulate('click')
		expect(fn).toBeCalled()
	})
})
