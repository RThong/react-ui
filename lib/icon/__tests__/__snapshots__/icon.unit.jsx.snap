// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon 渲染 传入props 1`] = `
<i
  className="hong-icon"
  type="alipay"
>
  <svg>
    <use
      xlinkHref="#alipay"
    />
  </svg>
</i>
`;

exports[`icon 渲染 未传入props 1`] = `
<i
  className="hong-icon"
>
  <svg>
    <use
      xlinkHref="#undefined"
    />
  </svg>
</i>
`;
